.github/workflows/test.yml

```
@@ -0,0 +1,153 @@
+name: Tests
+on:
+  push:
+    branches:
+      - develop
+  pull_request:
+    branches-ignore:
+      - master
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.ref }}
+  cancel-in-progress: true
+
+env:
+  DEFAULT_PYTHON: '3.12'
+
+jobs:
+  tests:
+    name: Run ${{ matrix.mark }} tests (Python ${{ matrix.python-version }}, Pydantic v${{ matrix.pydantic-version }} on ${{ matrix.os }})
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ['3.7', '3.12']
+        pydantic-version: [1, 2]
+        os: [ubuntu-latest]
+    env:
+      POETRY_VERSION: ${{ matrix.python-version == '3.7' && '1.5.1' || '1.7.1' }}
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Build Backend Container
+        uses: docker/build-push-action@v5
+        with:
+          tags: mtsrus/horizon-backend:${{ github.sha }}
+          context: .
+          file: docker/Dockerfile.backend
+          push: false
+          load: true
```

```suggestion
      - name: Build Backend Container
        uses: docker/build-push-action@v5
        with:
          tags: mtsrus/horizon-backend:${{ github.sha }}
          context: .
          file: docker/Dockerfile.backend
          push: false
          load: true
          cache-to: type=gha,mode=max
          cache-from: type=gha
```

Кажется без этих параметров Github не кэширует образы, с собирает их с нуля